llama_toolchain/inference/adapters/tgi/tgi.py

```diff
@@ -4,80 +4,51 @@
 # This source code is licensed under the terms described in the LICENSE file in
 # the root directory of this source tree.
 
-from typing import AsyncGenerator, List
 
-import httpx
+from typing import AsyncGenerator
 
+from huggingface_hub import InferenceClient
 from llama_models.llama3.api.chat_format import ChatFormat
-
-from llama_models.llama3.api.datatypes import Message, StopReason
+from llama_models.llama3.api.datatypes import StopReason
 from llama_models.llama3.api.tokenizer import Tokenizer
 
-from text_generation import Client
-
-from llama_toolchain.inference.api import *  # noqa: F403
+from llama_toolchain.inference.api import *
+from llama_toolchain.inference.api.api import (  # noqa: F403
+    ChatCompletionRequest,
+    ChatCompletionResponse,
+    ChatCompletionResponseStreamChunk,
+)
 from llama_toolchain.inference.prepare_messages import prepare_messages
 
+from .config import TGIImplConfig
 
-SUPPORTED_MODELS = {
+HF_SUPPORTED_MODELS = {
     "Meta-Llama3.1-8B-Instruct": "meta-llama/Meta-Llama-3.1-8B-Instruct",
     "Meta-Llama3.1-70B-Instruct": "meta-llama/Meta-Llama-3.1-70B-Instruct",
     "Meta-Llama3.1-405B-Instruct": "meta-llama/Meta-Llama-3.1-405B-Instruct",
 }
 
 
-class TGIInferenceAdapter(Inference):
-    def __init__(self, url: str) -> None:
-        self.url = url.rstrip("/")
+class TGIAdapter(Inference):
+
+    def __init__(self, config: TGIImplConfig) -> None:
+        self.config = config
         self.tokenizer = Tokenizer.get_instance()
         self.formatter = ChatFormat(self.tokenizer)
-        self.model = None
-        self.max_tokens = None
-
-    async def initialize(self) -> None:
-        hf_models = {v: k for k, v in SUPPORTED_MODELS.items()}
-
-        try:
-            print(f"Connecting to TGI server at: {self.url}")
-            async with httpx.AsyncClient() as client:
-                response = await client.get(f"{self.url}/info")
-                response.raise_for_status()
-                info = response.json()
-                if "model_id" not in info:
-                    raise RuntimeError("Missing model_id in model info")
-                if "max_total_tokens" not in info:
-                    raise RuntimeError("Missing max_total_tokens in model info")
-                self.max_tokens = info["max_total_tokens"]
-
-                model_id = info["model_id"]
-                if model_id not in hf_models:
-                    raise RuntimeError(
-                        f"TGI is serving model: {model_id}, use one of the supported models: {','.join(hf_models.keys())}"
-                    )
 
-                self.model = hf_models[model_id]
-        except Exception as e:
-            import traceback
+    @property
+    def client(self) -> InferenceClient:
+        return InferenceClient(base_url=self.config.url, token=self.config.api_token)
```

Comment:
Just want to make sure this works if no api_token is specified for the local TGI container

Reply:
Hi @ashwinb, The default value of `api_token` is set to an empty string when it's not specified and it's working fine with a local TGI container. However, I changed the default value to `None` to be able to retrieve the token from cache or local , works fine also with a local endpoint.